src/types.rs

```diff
@@ -127,6 +129,14 @@ pub struct AppleSubscriptionInfo {
     pub renewal_date: i64,
 }
 
+#[cfg(feature = "server")]
+#[derive(Clone, Debug, Deserialize, Serialize, Default)]
+pub struct ReferralInfo {
+    pub code: String,
+    pub pending_free_month: bool,
```

Comment:
It's surprising that we are tracking this separately? Wouldn't we just trigger to "top-up" as soon as this happens and forget about it other than incrementing the `received_free_months` counter?

Reply:
I may have chosen a bad name. `pending_free_month` is supposed to be `true` if the user hasn't paid yet but used a referral link to create the account, so we can show matching copy.

Reply:
Maybe `has_referrer`? This is maybe a bit weird if we unset it after purchase but since we will clean out the DB it probably makes sense. But most importantly it is suggesting that we should have doc comments. In fact we should probably add doc comments to basically all new fields unless they are particularly obvious. For example `code` could probably say "The code that this user can use to refer other users" to make it clear if it is incoming or outgoing as well.

This is also making me further question the usefulness of this object. Since this field is "referree data" whereas the others are "referrer data". Maybe we should just flatten them? I think we should avoid hierarchy unless there is a good reason for it.